Essentials/src/com/earth2me/essentials/commands/Commandmute.java

```diff
@@ -40,25 +40,32 @@ public void run(final Server server, final CommandSource sender, final String co
                 throw new Exception(tl("muteExempt"));
             }
         }
+
+        long muteTimestamp = 0;
+        String muteReason = null;
+
+        if (args.length > 1) {
+            final String time = args[1];
+            try {
+                muteTimestamp = DateUtil.parseDateDiff(time, true);
+                muteReason = getFinalArg(args, 2);
+            } catch (Exception e) {
+                muteReason = getFinalArg(args, 1);
+            }
+        } else {
+            user.setMuted(!user.getMuted());
```

Comment:
The event is Cancellable, so I don't think you should be doing this before knowing the outcome. I had a similar issue while implementing #2807 (which perhaps we should merge before handling this issue, since it seems a lot of the code refactoring is already done there).

Reply:
Whoops, that code shouldn't be up there. As for #2807, yeah that should be merged first